src/c++/library/grpc_client.cc

```diff
@@ -1091,13 +1091,20 @@ InferenceServerGrpcClient::PreRunProcessing(
   infer_request_.set_id(options.request_id_);
 
   infer_request_.mutable_parameters()->clear();
-  if (options.sequence_id_ != 0) {
+  if ((options.sequence_id_ != 0)) {
     (*infer_request_.mutable_parameters())["sequence_id"].set_int64_param(
         options.sequence_id_);
     (*infer_request_.mutable_parameters())["sequence_start"].set_bool_param(
         options.sequence_start_);
     (*infer_request_.mutable_parameters())["sequence_end"].set_bool_param(
         options.sequence_end_);
+  } else if (options.sequence_id_str_ != "") {
+    (*infer_request_.mutable_parameters())["sequence_id"].set_string_param(
+        options.sequence_id_str_);
```

Comment:
Only sequence_id needs to be in the if-then-else

Reply:
Wil do this. I thought less checks is faster, but I agree more checks is cleaner.